lib/logic/child_rewards_cubit.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/widgets.dart';
+import 'package:fokus/logic/auth/auth_bloc/authentication_bloc.dart';
+import 'package:fokus/model/currency_type.dart';
+import 'package:fokus/model/db/date/time_date.dart';
+import 'package:fokus/model/db/gamification/child_reward.dart';
+import 'package:fokus/model/db/gamification/points.dart';
+import 'package:fokus/model/db/gamification/reward.dart';
+import 'package:fokus/model/db/user/user_role.dart';
+import 'package:fokus/model/ui/gamification/ui_currency.dart';
+import 'package:fokus/model/ui/gamification/ui_reward.dart';
+import 'package:fokus/model/ui/user/ui_child.dart';
+import 'package:get_it/get_it.dart';
+
+import 'package:fokus/model/ui/user/ui_user.dart';
+import 'package:fokus/logic/reloadable/reloadable_cubit.dart';
+import 'package:fokus/services/data/data_repository.dart';
+import 'package:mongo_dart/mongo_dart.dart';
+
+class ChildRewardsCubit extends ReloadableCubit {
+	final ActiveUserFunction _activeUser;
+	final AuthenticationBloc _authBloc;
+	final DataRepository _dataRepository = GetIt.I<DataRepository>();
+
+  ChildRewardsCubit(this._activeUser, ModalRoute pageRoute, this._authBloc) : super(pageRoute);
+
+  void doLoadData() async {
+	  var child = _activeUser();
+		var caregiver = await _dataRepository.getUser(connected: child.id, role: UserRole.caregiver);
```

Comment:
You only use the `id` so why not just use the active user connections? No query needed

Reply:
So I need to implement connections to UIChild? (Making sure I'm doing something wrong again 😋)

Reply:
Or maybe move connections part of the UICaregiver to UIUser?

Reply:
Yes, both child and caregiver have connections so I'm not sure why they ended up in `UICaregiver` instead of `UIUser`, move them